src/connmgr/connection.rs

```diff
@@ -2110,14 +2110,11 @@ where
                 let max_read = cmp::min(tmp_buf.len(), zsess_in.credits() as usize);
 
                 let (size, mut finished) = match resp_body.try_recv(&mut tmp_buf[..max_read])? {
-                    RecvStatus::Complete(finished, written) => (written, Some(finished)),
-                    RecvStatus::Read((), written) => {
-                        if written == 0 {
-                            add_to_buffer.set(Some(resp_body.add_to_buffer()));
-                            continue;
-                        }
-
-                        (written, None)
+                    RecvStatus::Complete(finished, size) => (size, Some(finished)),
+                    RecvStatus::Read((), size) => (size, None),
+                    RecvStatus::NeedBytes(()) => {
+                        add_to_buffer.set(Some(resp_body.add_to_buffer()));
+                        continue;
                     }
                 };
```

Comment:
This is similar to the code that reads request body bytes from a client into a temporary buffer that is guaranteed to be non-zero-sized, except this time it's for reading response body bytes from a server. The updates have identical behavior as before so there should be no performance impact.